spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkScanBuilder.java

```diff
@@ -381,7 +381,7 @@ private Schema calculateMetadataSchema(List<Types.NestedField> metaColumnFields)
     AtomicInteger nextId = new AtomicInteger();
     return new Schema(
         metaColumnFields,
-        table.schema().identifierFieldIds(),
```

Comment:
Nit: I think for consistency with other places we should do `ImmutableSet.of()`, but it's nbd from my side.